libs/vkd3d/vkd3d_private.h

```diff
@@ -403,12 +403,88 @@ static inline void vkd3d_private_store_destroy(struct vkd3d_private_store *store
     pthread_mutex_destroy(&store->mutex);
 }
 
+static inline HRESULT vkd3d_private_data_lock(struct vkd3d_private_store *store)
+{
+    int rc;
+    if ((rc = pthread_mutex_lock(&store->mutex)))
+    {
+        ERR("Failed to lock mutex, error %d.\n", rc);
+        return hresult_from_errno(rc);
+    }
+
+    return S_OK;
+}
+
+static inline void vkd3d_private_data_unlock(struct vkd3d_private_store *store)
+{
+    pthread_mutex_unlock(&store->mutex);
+}
+
 HRESULT vkd3d_get_private_data(struct vkd3d_private_store *store,
         const GUID *tag, unsigned int *out_size, void *out);
-HRESULT vkd3d_set_private_data(struct vkd3d_private_store *store,
-        const GUID *tag, unsigned int data_size, const void *data);
-HRESULT vkd3d_set_private_data_interface(struct vkd3d_private_store *store,
-        const GUID *tag, const IUnknown *object);
+
+HRESULT vkd3d_private_store_set_private_data(struct vkd3d_private_store *store,
+        const GUID *tag, const void *data, unsigned int data_size, bool is_object);
+
+static inline HRESULT vkd3d_set_private_data(struct vkd3d_private_store *store,
+        const GUID *tag, unsigned int data_size, const void *data)
+{
+    return vkd3d_private_store_set_private_data(store, tag, data, data_size, false);
+}
+
+static inline HRESULT vkd3d_set_private_data_interface_locked(struct vkd3d_private_store *store,
```

Comment:
This is a bit confusing, since in other parts of the code a "_locked" suffix means that the function is an inner function that assumes the object is already locked. Probably shouldn't have any suffix here at all if we always expect it to be locked here.

Reply:
Makes sense, I'll swap this naming around.